Pull Request for Full Review

PlagueHO opened the pull request:
PR for performing a full module review.

See: https://github.com/PowerShell/DscResources/issues/145

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/plagueho/systemlocaledsc/2)

<!-- Reviewable:end -->

PlagueHO committed:
First commit

PlagueHO committed:
First complete version

PlagueHO committed:
Display available cultures in test

PlagueHO committed:
Change en-UK to en-AU because AppVeyor doesn't support en-UK

PlagueHO committed:
Fix Integration tests on AppVeyor

PlagueHO committed:
Fix Integration tests

PlagueHO committed:
Updated badge

PlagueHO committed:
Set Version number to 1.0.0.0

PlagueHO committed:
Examples fixed

PlagueHO committed:
Trimmed down Pester install code

PlagueHO committed:
Converted over to SystemLocaleDsc

PlagueHO committed:
Converted over to SystemLocaleDsc

PlagueHO committed:
Rename Example folder to Examples to meet guidelines

PlagueHO committed:
Merge branch 'dev' into reviewTarget

kwirkykat commented on DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1:
```diff
@@ -0,0 +1,148 @@
+    (
+        [parameter(Mandatory = $true)]
+        [ValidateSet('Yes')]
+        [String] $IsSingleInstance,
```
Why is this parameter here?
Value will always be 'Yes', and I don't see it used anywhere?

PlagueHO replied:
This is the standard single instance design pattern Travis came up with :smiley: 
https://blogs.msdn.microsoft.com/powershell/2015/06/22/want-to-write-a-dsc-resource-where-only-a-single-instance-can-be-configured/

https://msdn.microsoft.com/en-us/powershell/dsc/singleinstance

kwirkykat replied:
Ah. I see. All good then :+1:

kwirkykat commented on DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1:
```diff
@@ -0,0 +1,148 @@
+{
+    $Culture = $PSUICulture
+}
+Import-LocalizedData `
```
Importing the localized data is going to cause problems. Since only a 'en-us' strings file is currently provided, the resource may crash on any non-English system. See [this issue](https://github.com/PowerShell/xPSDesiredStateConfiguration/issues/168) from xPSDesiredStateConfiguration.

We worked around this issue in that module by simply commenting out this line and providing the localized data at the top of the file instead.

PlagueHO replied:
Good catch!

As an alternative, we worked around the problem in xNetworking this way:
https://github.com/PowerShell/xNetworking/blob/dev/DSCResources/MSFT_xDnsClientGlobalSetting/MSFT_xDnsClientGlobalSetting.psm1#L1

This allowed us to keep the strings separate and still work on non en-us systems. Would this alternative solution work?

kwirkykat replied:
Yes, that is probably an even better workaround.

PlagueHO replied:
Done!

kwirkykat commented:
@PlagueHO Can you hook this repo up to Reviewable?
https://reviewable.io/repositories#

kwirkykat commented:
Reviewed 3 of 12 files at r1.
Review status: 5 of 12 files reviewed at latest revision, 11 unresolved discussions.

---

*[appveyor.yml, line 10 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO7JR5b0wfX0BnrEY0U:-KO7JR5b0wfX0BnrEY0V:1162353606) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/appveyor.yml#L10)):*
> ```YAML
>         cd DscResource.Tests
>         Import-Module .\TestHelper.psm1 -force
>         Pop-Location
> ```

Is the change of location needed here?
You could just import the module from the original location with 
```powershell
Import-Module .\DscResource.Tests\TestHelper.psm1 -Force
```

---

*[LICENSE, line 21 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO7K2nC21cnvPLjs-Ky:-KO7K2nDfhWtA8CuLc2y:-1780283370) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/LICENSE#L21)):*
> ```
> LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
> OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
> THE SOFTWARE.
> ```

Missing a newline

---

*[README.md, line 10 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO7KJeetR5rLVcclEmR:-KO7KJeetR5rLVcclEmS:1589836791) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/README.md#L10)):*
> ```Markdown
> 
> If the System Locale is changed by this resource, it will require the node to reboot.
> If the LCM is not configured to allow restarting, the configration will not be able to be
> ```

Missing end of sentence?

---

*[README.md, line 11 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO7KaTd3GQeJHtwxqp8:-KO7KaTd3GQeJHtwxqp9:1074298691) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/README.md#L11)):*
> ```Markdown
> If the System Locale is changed by this resource, it will require the node to reboot.
> If the LCM is not configured to allow restarting, the configration will not be able to be
> ```

Please add this snippet about the Microsoft code of conduct (You can get the raw text with the links correctly formatted out of the top of the [contributing guidelines](https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md)) :
"This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments."

---

*[README.md, line 41 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO7Lz-OqZt6XQ0HtE5T:-KO7Lz-OqZt6XQ0HtE5U:-1605219138) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/README.md#L41)):*
> ```Markdown
> Configuration SetSystemLocale
> {
>    Param
> ```

'param' with a lowercase p please

---

*[Examples/SetSystemLocale.ps1, line 3 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO7N1I89Sj8vQOp3x6r:-KO7N1I89Sj8vQOp3x6s:-1605219138) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/Examples/SetSystemLocale.ps1#L3)):*
> ```PowerShell
> Configuration SetSystemLocale
> {
>    Param
> ```

'param' with a lowercase p please

---

*[Tests/Integration/MSFT_SystemLocale.config.ps1, line 1 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO7R9HxxEWUXnnCnVub:-KO7R9HxxEWUXnnCnVuc:349637520) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/Tests/Integration/MSFT_SystemLocale.config.ps1#L1)):*
> ```PowerShell
> $TestSystemLocale = [PSObject]@{
> ```

Just wondering: why cast this to a PSObject?

---

*[Tests/Integration/MSFT_SystemLocale.Integration.Tests.ps1, line 36 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO7S2PMondhGNCQyH4I:-KO7S2PNfY1QHQpUYQep:-152257556) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/Tests/Integration/MSFT_SystemLocale.Integration.Tests.ps1#L36)):*
> ```PowerShell
>         It 'Should compile without throwing' {
>             {
>                 Invoke-Expression -Command "$($Global:DSCResourceName)_Config -OutputPath `$TestEnvironment.WorkingFolder"
> ```

If you use an ampersand here instead of Invoke-Expression, you don't have to use awkward backticks in the arguments.
Like this:
```powershell
& "$($global:dscResourceName)_Config" -OutputPath $TestEnvironment.WorkingFolder
```

---

*[Tests/Unit/MSFT_SystemLocale.Tests.ps1, line 104 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO7UElsLdmvC0VDb5Mo:-KO7UEltaFFjGPRkrPBO:1548179938) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/Tests/Unit/MSFT_SystemLocale.Tests.ps1#L104)):*
> ```PowerShell
>             }
> 
>             It 'Should return false when Test is passed Time Zone that is not set' {
> ```

Time Zone --> System Locale ?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2)*
<!-- Sent from Reviewable.io -->

PlagueHO committed:
Fixes as per PR comments

PlagueHO commented:
Review status: 2 of 12 files reviewed at latest revision, 11 unresolved discussions.

---

*[appveyor.yml, line 10 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO7JR5b0wfX0BnrEY0U:-KO9-Q60_Isas_ty3pPY:1284231390) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/appveyor.yml#L10)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Is the change of location needed here?
> You could just import the module from the original location with 
> ```powershell
> Import-Module .\DscResource.Tests\TestHelper.psm1 -Force
> ```
</details>
Good point. Fixed.

---

*[LICENSE, line 21 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO7K2nC21cnvPLjs-Ky:-KO9-YZVdwwteSongJom:-499146009) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/LICENSE#L21)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Missing a newline
</details>
Done.

---

*[README.md, line 10 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO7KJeetR5rLVcclEmR:-KO9-lvL0BPdwDNp_i8M:-499146009) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/README.md#L10)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Missing end of sentence?
</details>
Done.

---

*[README.md, line 11 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO7KaTd3GQeJHtwxqp8:-KO90PAWkXlxxdeHAVje:-499146009) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/README.md#L11)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Please add this snippet about the Microsoft code of conduct (You can get the raw text with the links correctly formatted out of the top of the [contributing guidelines](https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md)) :
> "This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
> For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments."
> 
> 
</details>
Done.

---

*[README.md, line 41 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO7Lz-OqZt6XQ0HtE5T:-KO90Q00xvi-fTUZ6xyI:-499146009) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/README.md#L41)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> 'param' with a lowercase p please
</details>
Done.

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 7 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO6o_LnznoxIhESezWK-r1-7:-KO8yrjS7Gv4YsNK2Xz5:-499146009) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L7)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary>
> Done!
</details>
Done.

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 71 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO6pGH47ZNpomfEyq5p:-KO97SusljpJOjg0dcZz:1409586155) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L71)):*
<details><summary><i>Previously, TravisEz13 (Travis Plunk) wrote…</i></summary>
> What happens (what does the error look like) if the locale is not valid?
</details>
A very good point. If you pass Set-WinSystemLocale an invalid culture it doesn't actually return an error. So, I've made a change to Test-TargetResource so that it will validate the SystemLocale set in the parameter. If it is invalid it will throw a custom exception. I've also added new Unit tests for this.

---

*[Examples/SetSystemLocale.ps1, line 3 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO7N1I89Sj8vQOp3x6r:-KO8zBQkgzwFufPE8_JD:-499146009) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/Examples/SetSystemLocale.ps1#L3)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> 'param' with a lowercase p please
</details>
Done.

---

*[Tests/Integration/MSFT_SystemLocale.config.ps1, line 1 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO7R9HxxEWUXnnCnVub:-KO8zUSs_EOG7Yg_qxrL:-585414134) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/Tests/Integration/MSFT_SystemLocale.config.ps1#L1)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Just wondering: why cast this to a PSObject?
</details>
No good reason. Have removed it.

---

*[Tests/Integration/MSFT_SystemLocale.Integration.Tests.ps1, line 36 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO7S2PMondhGNCQyH4I:-KO8zrNkIlYJmm2wbkv0:1493611829) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/Tests/Integration/MSFT_SystemLocale.Integration.Tests.ps1#L36)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> If you use an ampersand here instead of Invoke-Expression, you don't have to use awkward backticks in the arguments.
> Like this:
> ```powershell
> & "$($global:dscResourceName)_Config" -OutputPath $TestEnvironment.WorkingFolder
> ```
</details>
An excellent point. Have changed. This actually comes from the Integration Test template. We should make a change to that as well.

---

*[Tests/Unit/MSFT_SystemLocale.Tests.ps1, line 104 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO7UElsLdmvC0VDb5Mo:-KO9-4br_AszidMyEOwa:156607381) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/Tests/Unit/MSFT_SystemLocale.Tests.ps1#L104)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Time Zone --> System Locale ?
</details>
Copy/Paste issue -good catch! Fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
@kwirkykat, @travisez13- all changes made. Should be good to go. Thank you both for reviewing!

kwirkykat commented:
Reviewed 1 of 12 files at r1, 7 of 9 files at r2.
Review status: 10 of 12 files reviewed at latest revision, 3 unresolved discussions.

---

*[README.md, line 41 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO7Lz-OqZt6XQ0HtE5T:-KOCHYKgHc8ojN2nReq5:536683174) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/README.md#L41)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary>
> Done.
</details>
This change didn't make it in?

---

*[Tests/Integration/MSFT_SystemLocale.Integration.Tests.ps1, line 36 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO7S2PMondhGNCQyH4I:-KOCH9Dq4ebd_20ojfLA:1005319513) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/Tests/Integration/MSFT_SystemLocale.Integration.Tests.ps1#L36)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary>
> An excellent point. Have changed. This actually comes from the Integration Test template. We should make a change to that as well.
</details>
Thanks for updating the template as well :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2)*
<!-- Sent from Reviewable.io -->

PlagueHO committed:
Fixes as per PR.

PlagueHO commented:
Review status: 10 of 12 files reviewed at latest revision, 3 unresolved discussions.

---

*[README.md, line 41 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO7Lz-OqZt6XQ0HtE5T:-KOEWs5aCiKLU6LYcwMp:665125567) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/README.md#L41)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> This change didn't make it in?
</details>
Doh! Made the change the Example file but not the readme. Done.

---

*[Tests/Integration/MSFT_SystemLocale.Integration.Tests.ps1, line 36 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO7S2PMondhGNCQyH4I:-KOEWXmrot94KMpQMWXS:1704463345) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/Tests/Integration/MSFT_SystemLocale.Integration.Tests.ps1#L36)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Thanks for updating the template as well :)
</details>
No problem :smile:

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
@kwirkykat - all done. Thank you!

kwirkykat commented:
Reviewed 1 of 1 files at r3.
Review status: 11 of 12 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Review status: 11 of 12 files reviewed at latest revision, 2 unresolved discussions.

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 71 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO6pGH47ZNpomfEyq5p:-KOkxXr_LS15YxkQe6PF:b-9330wc) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L71)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary>
> A very good point. If you pass Set-WinSystemLocale an invalid culture it doesn't actually return an error. So, I've made a change to Test-TargetResource so that it will validate the SystemLocale set in the parameter. If it is invalid it will throw a custom exception. I've also added new Unit tests for this.
</details>
@travisez13 - this should be ready for you to review.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Review status: 11 of 12 files reviewed at latest revision, 2 unresolved discussions.

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 7 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO6o_LnznoxIhESezWK-r1-7:-KOkxmtta5DF85stQlK7:b-suy5cy) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L7)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary>
> Done.
</details>
Hi @kwirkykat - Can this one be marked as resolved?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
Reviewed 1 of 9 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
Reviewed 1 of 12 files at r1.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 7 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO6o_LnznoxIhESezWK-r1-7:-KOugm7F3-YFwwQugTri:b8pc04l) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L7)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary>
> Hi @kwirkykat - Can this one be marked as resolved?
</details>
Seems like there's a bug in Reviewable. I keep acknowledging this and then I come back and it undoes itself...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
LGTM :+1:
@mbreakey3 Please review

mbreakey3 commented:
Reviewed 3 of 12 files at r1, 3 of 9 files at r2.
Review status: all files reviewed at latest revision, 18 unresolved discussions, all commit checks successful.

---

*[README.md, line 1 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvNQD5WgZO07rcFTNS:-KOvNQD5WgZO07rcFTNT:b8ncp2v) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/README.md#L1)):*
> ```Markdown
> [![Build status](https://ci.appveyor.com/api/projects/status/6fc4k4ef9isqsu6m/branch/dev?svg=true)](https://ci.appveyor.com/project/PlagueHO/systemlocaledsc/branch/dev)
> ```

Should this be switched to the master branch? Or will you change it once the module has been created in the DSC resource kit?

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 18 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvOvZbVnaEnGWmizSi:-KOvOvZbVnaEnGWmizSj:b-8fljfn) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L18)):*
> ```PowerShell
> }
> #endregion
> ```

Can you add [help comments](https://technet.microsoft.com/en-us/magazine/hh500719.aspx) for each of the functions (helper and exported)? .SYNOPSIS and .PARAMETERS are the only required attributes - add in any others as needed

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 39 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvOFLbW_-1pVVKH5Rz:-KOvOFLbW_-1pVVKH5S-:b-xnqh0w) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L39)):*
> ```PowerShell
> 
>     # Get the current System Locale
>     $CurrentSystemLocale = Get-WinSystemLocale `
> ```

lowercase: $currentSystemLocale

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 43 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvOJQOh-Yu_YRimWuh:-KOvOJQOh-Yu_YRimWui:b-t6c9ht) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L43)):*
> ```PowerShell
> 
>     # Generate the return object.
>     $ReturnValue = @{
> ```

lowercase: $returnValue

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 70 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvOPGPdnh_2TZqLIRZ:-KOvOPGPdnh_2TZqLIR_:b-n0lgac) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L70)):*
> ```PowerShell
> 
>     # Get the current System Locale
>     $CurrentSystemLocale = Get-WinSystemLocale `
> ```

lowercase

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 84 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvOZo2H4289-Ct6bqi:-KOvOZo2H4289-Ct6bqj:b-9a18fz) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L84)):*
> ```PowerShell
>             ) -join '' )
> 
>         $global:DSCMachineStatus = 1
> ```

Is this global declared somewhere else? Just wondering why it only appears here inside this function

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 99 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvPTnGpxuinjz8JEZs:-KOvPTnGpxuinjz8JEZt:bjd978h) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L99)):*
> ```PowerShell
>     param
>     (
>         [parameter(Mandatory = $true)]
> ```

Capitalize 'Parameter' (in all cases above and below as well)

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 121 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvOpyF00vTpObLuusY:-KOvOpyF00vTpObLuusZ:b-n0lgac) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L121)):*
> ```PowerShell
> 
>     # Get the current System Locale
>     $CurrentSystemLocale = Get-WinSystemLocale `
> ```

lowercase

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 162 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvPfO2vz2ct7B2Eldg:-KOvPfO2vz2ct7B2Eldh:b-w4l621) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L162)):*
> ```PowerShell
> 
> <#
> .SYNOPSIS
> ```

Indent the contents of the block comments

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 163 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvPpwoE7D7j6K_hJP4:-KOvPpwoE7D7j6K_hJP5:b-rf6812) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L163)):*
> ```PowerShell
> <#
> .SYNOPSIS
>  Checks the provided System Locale against the list of valid cultures.
> ```

Add the .PARAMETER attributes in this help comment as well

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 174 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvPwF4__WaqOW4QRBu:-KOvPwF4__WaqOW4QRBv:b-de4xin) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L174)):*
> ```PowerShell
>         [String] $SystemLocale
>     )
>     $ValidCultures = [System.Globalization.CultureInfo]::GetCultures(`
> ```

lowercase $validCultures

---

*[DSCResources/MSFT_SystemLocale/en-us/MSFT_SystemLocale.psd1, line 3 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvQ6vBpS8wrkd0rCWh:-KOvQ6vBpS8wrkd0rCWi:bmsohik) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/en-us/MSFT_SystemLocale.psd1#L3)):*
> ```PowerShell
> # culture="en-US"
> ConvertFrom-StringData -StringData @'
>     GettingSystemLocaleMessage=Getting Windows System Locale.
> ```

not to be super picky, but can you add spaces around the '=' signs

---

*[Examples/SetSystemLocale.ps1, line 1 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvQFrpVXypyybFAvr-:-KOvQFrpVXypyybFAvr0:biblqex) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/Examples/SetSystemLocale.ps1#L1)):*
> ```PowerShell
> Configuration SetSystemLocale
> ```

Could you add a brief description about the example? How/when to use it/what it does/etc.?

---

*[Tests/Integration/MSFT_SystemLocale.Integration.Tests.ps1, line 1 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvQXRA8Uq9lvsMcwwL:-KOvQXRBHjcQbA50tBYB:b-723821) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/Tests/Integration/MSFT_SystemLocale.Integration.Tests.ps1#L1)):*
> ```PowerShell
> $Global:DSCModuleName      = 'SystemLocaleDsc'
> ```

I know this was in our old template - but can you change these globals to be script variables instead:
$script:DSCModuleName      = 'SystemLocaleDsc'
$script:DSCResourceName    = 'MSFT_SystemLocale'

---

*[Tests/Integration/MSFT_SystemLocale.Integration.Tests.ps1, line 21 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvQrV-_-6jqS-9Zzyb:-KOvQrV-_-6jqS-9Zzyc:b-n0lgac) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/Tests/Integration/MSFT_SystemLocale.Integration.Tests.ps1#L21)):*
> ```PowerShell
> 
> # Store the test machine system locale
> $CurrentSystemLocale = (Get-WinSystemLocale).Name
> ```

lowercase

---

*[Tests/Integration/MSFT_SystemLocale.Integration.Tests.ps1, line 51 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvR-hpUXgOrfg5ier9:-KOvR-hpUXgOrfg5ierA:bos14m7) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/Tests/Integration/MSFT_SystemLocale.Integration.Tests.ps1#L51)):*
> ```PowerShell
>             }
>             # A reboot would need to occur before this node can be bought into alignment
>             # $current.SystemLocale     | Should Be $TestSystemLocale.SystemLocale
> ```

If this commented-out line isn't used it should be deleted 

---

*[Tests/Unit/MSFT_SystemLocale.Tests.ps1, line 1 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvR7kReXHb0SBlogFV:-KOvR7kReXHb0SBlogFW:bwfwz1e) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/Tests/Unit/MSFT_SystemLocale.Tests.ps1#L1)):*
> ```PowerShell
> $Global:DSCModuleName      = 'SystemLocaleDsc'
> ```

again - change to $script

---

*[Tests/Unit/MSFT_SystemLocale.Tests.ps1, line 25 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvRDJlQLShu3SW3ipn:-KOvRDJlQLShu3SW3ipo:b2noaab) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/Tests/Unit/MSFT_SystemLocale.Tests.ps1#L25)):*
> ```PowerShell
>     #region Pester Tests
> 
>     InModuleScope $Global:DSCResourceName {
> ```

'InModuleScope' was also in an old test template - now we're only using it around (non-exported helper functions)[https://github.com/PowerShell/DscResources/blob/master/Tests.Template/unit_template.ps1#L115] so the exported function tests (Get/Set/Test) should be moved out of this scope

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
Review status: all files reviewed at latest revision, 21 unresolved discussions.

---

*[SystemLocaleDsc.psd1, line 45 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvfJRnLnn6GvzmDUuQ:-KOvfJRnLnn6GvzmDUuR:b-nb2iy7) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/SystemLocaleDsc.psd1#L45)):*
> SystemLocale
SystemLocaleDsc

---

*[SystemLocaleDsc.psd1, line 48 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvfMwg_mp_Htb8m6FP:-KOvfMwg_mp_Htb8m6FQ:b-c81ron) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/SystemLocaleDsc.psd1#L48)):*
> SystemLocale
SystemLocaleDsc

---

*[SystemLocaleDsc.psd1, line 54 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvfPibadKm8g3cigv-:-KOvfPibadKm8g3cigv0:b-nb2iy7) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/SystemLocaleDsc.psd1#L54)):*
> SystemLocale
SystemLocaleDsc

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2)*
<!-- Sent from Reviewable.io -->

mbreakey3 commented:
*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 84 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvOZo2H4289-Ct6bqi:-KOvl0wMzzrjY3eXkcRe:b-yeenp6) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L84)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Is this global declared somewhere else? Just wondering why it only appears here inside this function
</details>
LGTM - Nevermind, this is fine

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Great review @mbreakey3 - I'll get onto this tonight after work.

PlagueHO committed:
Fixes as per PR comments

PlagueHO commented:
Ok @mbreakey3 , @kwirkykat  - all done! Thanks for reviewing! I'll try and go through the other repos I'm submitting and make sure they match up so you don't need as many review changes. The big changes though are the InModuleScope ones (as I'm doing xNetworking as well). They are actually require a significant amount of time to complete. But I'll try and give you a shout when they're done.

---

Reviewed 1 of 12 files at r1, 4 of 9 files at r2, 1 of 1 files at r3.
Review status: 5 of 12 files reviewed at latest revision, 21 unresolved discussions.

---

*[README.md, line 1 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvNQD5WgZO07rcFTNS:-KP0J005dgZ5vb0Zdosk:bv575vf) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/README.md#L1)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Should this be switched to the master branch? Or will you change it once the module has been created in the DSC resource kit?
</details>
Good point - fixed. You'll still need to change it once the repo is moved though.

---

*[SystemLocaleDsc.psd1, line 45 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvfJRnLnn6GvzmDUuQ:-KP0JIMInQG97PxH8u8p:b-896fix) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/SystemLocaleDsc.psd1#L45)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > SystemLocale
> SystemLocaleDsc
</details>
Done.

---

*[SystemLocaleDsc.psd1, line 48 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvfMwg_mp_Htb8m6FP:-KP0JIhvmILA0Ci2UymC:b-896fix) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/SystemLocaleDsc.psd1#L48)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > SystemLocale
> SystemLocaleDsc
> 
</details>
Done.

---

*[SystemLocaleDsc.psd1, line 54 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvfPibadKm8g3cigv-:-KP0JSSHKxIceg83g7Hv:b-896fix) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/SystemLocaleDsc.psd1#L54)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > SystemLocale
> SystemLocaleDsc
</details>
Done.

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 18 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvOvZbVnaEnGWmizSi:-KP0KTicG-klVhp7-iBL:b-896fix) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L18)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Can you add [help comments](https://technet.microsoft.com/en-us/magazine/hh500719.aspx) for each of the functions (helper and exported)? .SYNOPSIS and .PARAMETERS are the only required attributes - add in any others as needed
</details>
Done.

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 39 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvOFLbW_-1pVVKH5Rz:-KP0LAX2Rq7Jq2hRLPYA:b-896fix) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L39)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> lowercase: $currentSystemLocale
</details>
Done.

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 43 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvOJQOh-Yu_YRimWuh:-KP0LFKqjYNcP2SuuLGv:b-896fix) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L43)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> lowercase: $returnValue
</details>
Done.

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 70 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvOPGPdnh_2TZqLIRZ:-KP0LGXRIj0mJy4lVr4d:b-896fix) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L70)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> lowercase
</details>
Done.

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 84 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvOZo2H4289-Ct6bqi:-KP0LIRXCAYUH9aZrJL4:bpv1dcz) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L84)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> LGTM - Nevermind, this is fine
</details>
Hehe - I wish the DSC team could have used a more obvious name or method for triggering a reboot :smile: E.g. Call a method on the LCM to notify a resource requires a reboot. Maybe for DSCv3? :smile:

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 99 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvPTnGpxuinjz8JEZs:-KP0LjNgrO-LbNnFE-05:b-896fix) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L99)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Capitalize 'Parameter' (in all cases above and below as well)
</details>
Done.

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 121 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvOpyF00vTpObLuusY:-KP0M-PMWTG1mt1qF5hu:b-896fix) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L121)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> lowercase
</details>
Done.

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 162 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvPfO2vz2ct7B2Eldg:-KP0MOEA9chnfoE-WfY2:b-896fix) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L162)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Indent the contents of the block comments
</details>
Done.

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 163 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvPpwoE7D7j6K_hJP4:-KP0MP3yD5-dB8PFuC3_:b-896fix) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L163)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Add the .PARAMETER attributes in this help comment as well
</details>
Done.

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 174 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvPwF4__WaqOW4QRBu:-KP0MT4bZ7JDi8PdVJap:b-896fix) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L174)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> lowercase $validCultures
</details>
Done.

---

*[DSCResources/MSFT_SystemLocale/en-us/MSFT_SystemLocale.psd1, line 3 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvQ6vBpS8wrkd0rCWh:-KP0MovzznVoC4czoGLY:b-63zk9i) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/en-us/MSFT_SystemLocale.psd1#L3)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> not to be super picky, but can you add spaces around the '=' signs
</details>
No worries! I'm a big fan of consistency. Done.

---

*[Examples/SetSystemLocale.ps1, line 1 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvQFrpVXypyybFAvr-:-KP0PNt2ultw7kn8GsVD:b-896fix) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/Examples/SetSystemLocale.ps1#L1)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Could you add a brief description about the example? How/when to use it/what it does/etc.?
</details>
Done.

---

*[Tests/Integration/MSFT_SystemLocale.Integration.Tests.ps1, line 1 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvQXRA8Uq9lvsMcwwL:-KP0Q6P3-io7Q1Pl-zyn:b8mwcp9) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/Tests/Integration/MSFT_SystemLocale.Integration.Tests.ps1#L1)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> I know this was in our old template - but can you change these globals to be script variables instead:
> $script:DSCModuleName      = 'SystemLocaleDsc'
> $script:DSCResourceName    = 'MSFT_SystemLocale'
</details>
No worries (I wrote the original template so it is kind of my fault anyway :smile: ). Done.

---

*[Tests/Integration/MSFT_SystemLocale.Integration.Tests.ps1, line 21 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvQrV-_-6jqS-9Zzyb:-KP0S3dTjRuaGk9CFT2T:b-896fix) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/Tests/Integration/MSFT_SystemLocale.Integration.Tests.ps1#L21)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> lowercase
</details>
Done.

---

*[Tests/Integration/MSFT_SystemLocale.Integration.Tests.ps1, line 51 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvR-hpUXgOrfg5ier9:-KP0SBw7hK9ma3M4QvCn:b-896fix) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/Tests/Integration/MSFT_SystemLocale.Integration.Tests.ps1#L51)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> If this commented-out line isn't used it should be deleted 
</details>
Done.

---

*[Tests/Unit/MSFT_SystemLocale.Tests.ps1, line 1 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvR7kReXHb0SBlogFV:-KP0SCSYyhnew_BL_26m:b-896fix) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/Tests/Unit/MSFT_SystemLocale.Tests.ps1#L1)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> again - change to $script
</details>
Done.

---

*[Tests/Unit/MSFT_SystemLocale.Tests.ps1, line 25 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvRDJlQLShu3SW3ipn:-KP0SDf4P99Q5Q7EI5WX:b2e0x24) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/Tests/Unit/MSFT_SystemLocale.Tests.ps1#L25)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> 'InModuleScope' was also in an old test template - now we're only using it around (non-exported helper functions)[https://github.com/PowerShell/DscResources/blob/master/Tests.Template/unit_template.ps1#L115] so the exported function tests (Get/Set/Test) should be moved out of this scope
</details>
Done. I also corrected all the first letter capitilization of the variable names in this script as well (e.g. changed $TestSystemLocale to $testSystemLocale) I'll keep an eye out for these in future. Sorry about that.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-:-KP0ScciCq9lma3nxO_d:bqt6bug)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Reviewed 7 of 7 files at r4.
Review status: all files reviewed at latest revision, 21 unresolved discussions.

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 7 \[r1\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KO6o_LnznoxIhESezWK-r1-7:-KP0qcOmd56MtO6SxlaP:b59016g) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/171eb3e6be7140ac5534cd8961882b14532dc893/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L7)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Seems like there's a bug in Reviewable. I keep acknowledging this and then I come back and it undoes itself...
</details>
Oh yes! I'm having trouble with this bit as well. I had to refresh the page to even see this comment from you.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2)*
<!-- Sent from Reviewable.io -->

mbreakey3 commented:
*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 84 \[r3\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KOvOZo2H4289-Ct6bqi:-KPEtnjkoYABp1Qk_4D6:biurk9z) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/20de0a659c300a21502508235abef2e5e5521891/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L84)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary>
> Hehe - I wish the DSC team could have used a more obvious name or method for triggering a reboot :smile: E.g. Call a method on the LCM to notify a resource requires a reboot. Maybe for DSCv3? :smile:
</details>
Haha, agreed....

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2)*
<!-- Sent from Reviewable.io -->

mbreakey3 commented:
Reviewed 6 of 7 files at r4.
Review status: all files reviewed at latest revision, 22 unresolved discussions.

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 110 \[r4\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KPEu0BRWnM8jYF-4m_8:-KPEu0BRWnM8jYF-4m_9:bdgkllr) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/efe5c08af5530deeeed42c0a09ee73671314e47e/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L110)):*
> ```PowerShell
> <#
>     .SYNOPSIS
>     Tests if the current System Local on the node needs to be changed.
> ```

Should this be 'System Locale'?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2)*
<!-- Sent from Reviewable.io -->

PlagueHO committed:
Correct spelling of System Locale

PlagueHO commented:
Review status: 11 of 12 files reviewed at latest revision, 3 unresolved discussions.

---

*[DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1, line 110 \[r4\]](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-KPEu0BRWnM8jYF-4m_8:-KPH8hAjx9edJ_NKnp4i:brqnkrj) ([raw file](https://github.com/plagueho/systemlocaledsc/blob/efe5c08af5530deeeed42c0a09ee73671314e47e/DSCResources/MSFT_SystemLocale/MSFT_SystemLocale.psm1#L110)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> Should this be 'System Locale'?
</details>
Doh! Yes - it should. Fixed. And the other one.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2)*
<!-- Sent from Reviewable.io -->

mbreakey3 commented:
Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2)*
<!-- Sent from Reviewable.io -->

mbreakey3 commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-:-KPJUPhElst8s4jYmFFU:bnfp4nl)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
@PlagueHO 3 reviewers have approved! You can go ahead and close this PR now. Make sure all the changes you made are merged into your main master/dev branch(es) and then transfer the repo over to me (kwirkykat) when you're ready

PlagueHO commented:
Thanks @kwirkykat  - I'll get this merged etc and transfer today. Thanks again for all your (and @mbreakey3  and @travisez13 ) on this.

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2#-:-KPKLNbzVNXBYGEVVVpY:bmnotrc)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Reviewed 2 of 12 files at r1, 3 of 9 files at r2, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/systemlocaledsc/2)*
<!-- Sent from Reviewable.io -->
